### PR TITLE
Add some suggestions from issues

### PR DESCRIPTION
**Pull request by Sunketchupm**
Each commit is a different suggestion. Specifically, issues #343, #349, and #264. One of the suggestions doesn't come from an issue but still something I heard about and wanted to make.

### CODE REVIEW

**Comment by Cooliokid956** [comment]
is there a reason why the coopnet id function was limited to only the local player's id? the issue didn't narrow it down to only the local id

**Comment by Sunketchupm** [comment]
The reason is that this is based almost entirely on the get discord id function.
There's no real reason otherwise though, so I'll change that.

**Comment by Cooliokid956** [comment]
if you didn't notice, the original discord id function was undeprecated a while ago, now discord ids are networked

File: src/pc/lua/utils/smlua_misc_utils.c [hunk 1]
```diff
@@ -426,6 +426,14 @@ const char* get_local_discord_id(void) {
 #endif
 }
 
+const char* get_coopnet_id(s8 localIndex) {
```
Comment (Isaac0-dev) [inline src/pc/lua/utils/smlua_misc_utils.c:430]: this function will allow mods to see ip addresses if direct connection is used, which isn't really ideal.

File: src/game/interaction.c [hunk 2]
```diff
@@ -1327,10 +1328,12 @@ static u8 resolve_player_collision(struct MarioState* m, struct MarioState* m2)
...
 
 u8 determine_player_damage_value(u32 interaction) {
-    if (interaction & INT_GROUND_POUND_OR_TWIRL) { return 3; }
+    if (interaction & INT_GROUND_POUND) { return 4; }
+    if (interaction & (INT_TWIRL | INT_PUNCH | INT_TRIP)) { return 3; }
     if (interaction & INT_KICK) { return 2; }
-    if (interaction & INT_ATTACK_SLIDE) { return 1; }
-    return 2;
+    if (interaction & INT_SLIDE_KICK) { return 2; }
+    if (interaction & INT_FAST_ATTACK_OR_SHELL) { return 1; }
```
Comment (Isaac0-dev) [inline src/game/interaction.c]: this line does nothing

**Review by Isaac0-dev — CHANGES_REQUESTED** [review]

File: src/game/interaction.c (hunk 2 above)
Comment (Sunketchupm) [reply src/game/interaction.c]: Which line?

Comment (Isaac0-dev) [reply src/game/interaction.c]: `if (interaction & INT_FAST_ATTACK_OR_SHELL) { return 1; }`

**Comment by Sunketchupm** [comment]
Surprised no one mentioned the pvp damage changes that were suggested. I talked with saus and theman, both of whom do a lot of pvp, and they think this is good. However I would like some other opinions on it since it will change pvp a lot.

**Comment by EmeraldLoc** [comment]
PvP changes are generally good. Punches should 100% do more damage, so it’s good it does that now. The only thing that I feel like could be iffy is the slide kick, but that’ll take some tinkering. Also I don’t know how I feel about the dive thing, while it does make sense, dives having good kickback feels pretty broken, once again takes some testing.

**Pull request merged**